src/controllers/userController.ts

```diff
@@ -0,0 +1,67 @@
+import { Response, Request } from 'express';
+import MyCustomError from '../utils/customError';
+import { catchAsync } from '../middlewares/error';
+import { userService } from '../services/userService';
+import { userDTO } from '../dto/user.dto';
+
+let certificationNumber: string;
+
+class userController {
+  constructor(private Service: userService) {}
+
+  public checkEmail = catchAsync(async (req: Request, res: Response) => {
+    const email: string = req.body.email;
+
+    if (!email) throw new MyCustomError('이메일을 확인해 주세요.', 400);
+
+    await this.Service.checkEmail(email);
+    return res.status(200).json({ message: '사용 가능한 이메일입니다.' });
+  });
+
+  public sendEmail = catchAsync(async (req: Request, res: Response) => {
+    const email: string = req.body.email;
+    certificationNumber = Math.random().toString().substring(2, 8);
+
+    await this.Service.emailSend(email, certificationNumber);
+    return res.status(200).json({ message: '인증 이메일이 발송 되었습니다.' });
+  });
+
+  public confirmEmail = catchAsync(async (req: Request, res: Response) => {
+    const userEmailCode: string = req.body.userEmailCode;
```

Comment:
```suggestion
  public confirmEmail = catchAsync(async (req: Request<{}, {}, {userEmailCode: string}, {}>, res: Response) => {
    const {userEmailCode} = req.body;
```
두 코드 모두 기능상으로는 동일합니다. 그러나 더 안전한 타입스크립트 코딩을 위해서는, 두 번째 코드가 더 좋은 코드입니다.

첫 번째 코드에서는 Request 인터페이스가 일반적인 형식이므로, 모든 요청이 이 형식에 맞는지 확인할 수 없습니다. 이 경우, 유효성 검사를 수행하고 타입을 검사하는 데 더 많은 작업이 필요합니다.

두 번째 코드에서는 Request 인터페이스가 제네릭 형식으로 선언되어, 요청 유형이 명확하게 정의됩니다. 이를 통해 코드의 안정성을 높일 수 있습니다. 또한 TypeScript의 유효성 검사를 통해 컴파일 시점에서 오류를 검출할 수 있습니다.